### PR TITLE
fix bug when startKey is nil or endKey is nil when use ListRegionIDsI…

### PR DESCRIPTION
**Pull request by zhengcf**
fix bug when startKey is nil or endKey is nil when use ListRegionIDsInKeyRange only return first region

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -226,7 +226,7 @@ func (c *RegionCache) ListRegionIDsInKeyRange(bo *Backoffer, startKey, endKey []
 			return nil, errors.Trace(err)
 		}
 		regionIDs = append(regionIDs, curRegion.Region.id)
-		if curRegion.Contains(endKey) {
+		if (startKey!=nil && curRegion.Contains(endKey)) || len(curRegion.EndKey) == 0 {
```
Comment (AndreMouche) [inline store/tikv/region_cache.go]: Is it possible that  startKey is not nil and curRegion.start_key is nil?

**Comment by ngaut** [comment]
Nice catch, please sign the CLA by click the "CLA not singed yet" button.

File: store/tikv/region_cache.go (hunk 1 above)
Comment (disksing) [reply store/tikv/region_cache.go]: I suppose it should be:
if (**endKey** != nil && curRegion.Contains(endKey) || len(curRegion.EndKey) == 0 {

Comment (coocood) [reply store/tikv/region_cache.go]: I think this is the bug of `Contains` method.
Fix it there is better.
If a region has nil end key, it contains every key.

**Comment by coocood** [comment]
@zhengcf 
Your email may not be set up correctly in your Git, so the commit shows the user `zcf`.

File: store/tikv/region_cache.go (hunk 1 above)
Comment (AndreMouche) [reply store/tikv/region_cache.go]: I suppose it should be:
```
if (endKey != nil && curRegion.Contains(endKey) ||
   (endKey==nil && len(curRegion.EndKey)==0 ) {
...
}
```

Comment (AndreMouche) [reply store/tikv/region_cache.go]: I do not agree with @coocood  since function `Contains` can not distinguish the key is `startKey` or `endKey`

**Comment by ngaut** [comment]
Ping @zhengcf

**Comment by coocood** [comment]
@zhengcf 
There is no update since the day PR is sent.
So I close this PR for now.
We can reopen it if you have any update later.

**Pull request closed without merging**